nomad/structs/structs_test.go

```diff
@@ -110,6 +110,16 @@ func TestJob_Validate(t *testing.T) {
 	if !strings.Contains(mErr.Errors[2].Error(), "Task group web validation failed") {
 		t.Fatalf("err: %s", err)
 	}
+
+	// test for empty datacenters
+	j = &Job{
+		Datacenters: []string{""},
+	}
+	err = j.Validate()
+	mErr = err.(*multierror.Error)
+	if !strings.Contains(mErr.Error(), "datacenter must be non-empty string") {
```

Comment:
You can use testify require here instead of using `Fatalf` directly

```suggestion
	require.Contains(t, mErr.Error(), "datacenter must be non-empty string")
```